watertap/unit_models/gac.py

```diff
@@ -760,243 +708,347 @@ def eq_dg(b, t, j):
+            return b.process_flow.properties_in[t].pressure == b.gac_removed[t].pressure
 
+        # mass transfer of target_species
+        # TODO: check for mass based (not mole) get_material_flow_terms, but ok under mcas_prop_pack
         @self.Constraint(
-            doc="Empty bed contact time of the mass transfer zone"
-            "at the time of replacement",
+            self.flowsheet().config.time,
+            self.target_species,
+            doc="mass transfer for adsorbed solutes in 'target_species' within 'gac_removed' (out of the bed)",
         )
-        def eq_ebct_mtz(b):
-            return (
-                b.ebct_mtz_replace
-                == (b.mass_throughput_mtz_upstream - b.mass_throughput) * b.min_ebct
+        def eq_mass_transfer_cv(b, t, j):
+            return (1 - b.conc_ratio_avg) * b.process_flow.properties_in[
+                t
+            ].get_material_flow_terms("Liq", j) == (
+                -b.process_flow.mass_transfer_term[t, "Liq", j]
             )
 
-        @self.Constraint(doc="Length of the mass transfer zone")
-        def eq_length_mtz(b):
-            return b.length_mtz_replace == b.velocity_sup * b.ebct_mtz_replace
-
+        # mass balance for port not in control volume
         @self.Constraint(
-            doc="Length of the mass transfer zone at the time of replacement"
+            self.flowsheet().config.time,
+            self.target_species,
+            doc="mass balance for port not in control volume",
         )
-        def eq_approx_saturation(b):
-            return (1 * (b.bed_length - b.length_mtz_replace)) + (
-                ((b.saturation_mtz_upstream + b.conc_ratio_replace) / 2)
-                * b.length_mtz_replace
-            ) == b.bed_length * b.gac_saturation_replace
+        def eq_mass_transfer_port(b, t, j):
+            return (
+                b.gac_removed[t].get_material_flow_terms("Liq", j)
+                == -b.process_flow.mass_transfer_term[t, "Liq", j]
+            )
+
+        # no mass transfer of inert_species, fix to 0 to avoid 0 solver tolerances
+        for j in self.inert_species:
+            self.process_flow.mass_transfer_term[:, "Liq", j].fix(0)
+            self.gac_removed[0].get_material_flow_terms("Liq", j).fix(0)
 
         # ---------------------------------------------------------------------
         if (
             self.config.film_transfer_coefficient_type
             == FilmTransferCoefficientType.calculated
         ):
-            # TODO check N_Re formulation for packed beds
+
             self.N_Re = Var(
-                initialize=1,
+                initialize=10,
                 bounds=(0, None),
                 domain=NonNegativeReals,
                 units=pyunits.dimensionless,
-                doc="Reynolds number, correlation using Schmidt number valid in Re < 2e4",
+                doc="Reynolds number, correlations using Reynolds number valid in Re < 2e4",
```

Comment:
Any reason to set the bounds for ``N_Re`` (and other dimensionless variables) to the validated ranges? I have done this in IX to make sure it stays in the assumptions for the model, but I'm not sure if its the best approach. I won't let it hold up the PR.

Reply:
This is the same logic as yours, basically N_Re should be laminar for the correlations to be valid. I see it from this perspective though. If a user is inputting variables into the model for a design and flow becomes turbulent one of two things will happen:

- the bound is not there, the model will solve but N_Re will be out of the assumptions range, it's on the user to identify the poor conditions
- the bound is there, the model will not solve

This may be troublesome and more tricky to debug when the model isn't solving because it's ambiguous if the error is with N_Re hitting a bound or something else with the solve. Could use more input here though.

Reply:
I personally tend to use the fitted range for the bounds relating to correlations, as this is the range over which we know the correlation is valid. If someone tries to exceed the bound, then there is no guarantee the result is correct even if the solver finds an "optimal" solution, so I tend to err towards a definitive failure rather than providing a potentially invalid answer. Users can always relax bounds later (at their own risk).

IDAES is trying to work on better diagnostics tools this year which would help users identify the source of an infeasible model.